### PR TITLE
add variable check before setting tidb_row_format_version

### PR DESCRIPTION
**Pull request by birdstorm**
### What problem does this PR solve? 
Fix CI error when using against old versions of tidb.

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala [hunk 1]
```diff
@@ -223,7 +223,10 @@ trait SharedSQLContext extends SparkFunSuite with Eventually with Logging with S
     _statement = _tidbConnection.createStatement()
     // TODO: support new row format
     //  https://github.com/pingcap/tispark/issues/1355
-    _statement.execute("SET GLOBAL tidb_row_format_version=1")
+    val resultSet = _statement.executeQuery("SHOW GLOBAL VARIABLES LIKE 'tidb_row_format_version'")
+    if (resultSet.next()) {
```
Comment (zhexuany) [inline core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala]: resultSet.hashNext()?

Comment (birdstorm) [reply core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala]: ResultSet use `next()` to indicate whether next row exists

**Comment by birdstorm** [comment]
Test failure due to #1357

**Pull request merged**